### PR TITLE
API:column_family.cc Add get_build_index implmentation

### PR DESCRIPTION
**Pull request by amnonh**
This Patch adds an implementation of the get build index API and remove a
FIXME.

The API returns a list of secondary indexes belongs to a column family.

Example:

CREATE KEYSPACE scylla_demo WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'};

CREATE TABLE scylla_demo.mytableID (     uid uuid,     text text,     time timeuuid,     PRIMARY KEY (uid, time) );
CREATE index on scylla_demo.mytableID (time);

$ curl -X GET 'http://localhost:10000/column_family/built_indexes/scylla_demo%3Amytableid'
["mytableid_time_idx"]

Signed-off-by: Amnon Heiman <amnon@scylladb.com>

### CODE REVIEW

**Comment by amnonh** [comment]
when I look at the code I didn't see any indication of the index that it's ready or not, so I assume we add index that are already built, I'll have another look, maybe  I missed it

**Comment by nyh** [comment]
@amnonh unfortunately I don't remember when we add the index to the list (maybe @penberg or @psarna remember), and quickly browsing through the code I couldn't find an answer. But clearly we need to remember an index immediately when it created - e.g., to forbid creating another index with the same name - so I thought it might be remembered immediately, even before being built. You can test this in practice - add an index to a fairly large data, and see if the index list listed a millisecond later or just a minute later (you can also see in the log messages when the MV build was complete).
Another question I don't have an answer for is whether it is really important that this API list only the "built" indexes. I assumed it is because of its name.

**Comment by psarna** [comment]
`list_indexes` will return created indexes, and it doesn't indicate in any way if the underlying materialized views are built or not. The only way to know for sure is to query `system_distributed.view_build_status` or `system.views_builds_in_progress` and reason from its output.

**Comment by amnonh** [comment]
Thanks @psarna I'll update accordingly

**Comment by amnonh** [comment]
@nyh I've forced push a new version after validating that it does not include indexes that are being built.

I added an additional clean up patch for an issue I've found during the implementation that remove unhandle future warning in the compilation

File: api/column_family.cc [hunk 1]
```diff
@@ -843,13 +848,28 @@ void set_column_family(http_context& ctx, routes& r) {
...
     });
 
-    cf::get_built_indexes.set(r, [](const_req) {
-        // FIXME
-        // Currently there are no index support
-        return std::vector<sstring>();
+    cf::get_built_indexes.set(r, [&ctx](std::unique_ptr<request> req) {
+        auto [ks, cf_name] = parse_fully_qualified_cf_name(req->param["name"]);
+        return db::system_keyspace::load_view_build_progress().then([ks, cf_name, &ctx](const std::vector<db::system_keyspace::view_build_progress>& vb) mutable {
+            std::set<sstring> vp;
```
Comment (nyh) [inline api/column_family.cc]: Why not std::unordered_set?
But I'll let it slide.

**Comment by nyh** [comment]
I committed your branch. When it reaches from next to master, if not rebased in the middle, this pull request should be closed automatically.

**Pull request merged**